services/dynamodb/src/it/java/software/amazon/awssdk/services/dynamodb/ScanPaginatorIntegrationTest.java

```diff
@@ -0,0 +1,199 @@
+import software.amazon.awssdk.services.dynamodb.model.KeySchemaElement;
+import software.amazon.awssdk.services.dynamodb.model.KeyType;
+import software.amazon.awssdk.services.dynamodb.model.ProvisionedThroughput;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.ScalarAttributeType;
+import software.amazon.awssdk.services.dynamodb.model.ScanRequest;
+import software.amazon.awssdk.services.dynamodb.model.ScanResponse;
+import software.amazon.awssdk.services.dynamodb.paginators.ScanPaginator;
+import utils.resources.tables.BasicTempTable;
+import utils.test.util.DynamoDBTestBase;
+import utils.test.util.TableUtils;
+
+public class ScanPaginatorIntegrationTest extends DynamoDBTestBase {
+
+    private static final String TABLE_NAME = BasicTempTable.TEMP_TABLE_NAME;
+    private static final String HASH_KEY_NAME = BasicTempTable.HASH_KEY_NAME;
+    private static final String ATTRIBUTE_FOO = "attribute_foo";
+    private static final int ITEM_COUNT = 19;
+
+    @BeforeClass
+    public static void setUpFixture() throws Exception {
+        DynamoDBTestBase.setUpTestBase();
+
+        dynamo.createTable(CreateTableRequest.builder().tableName(TABLE_NAME)
+                                             .keySchema(KeySchemaElement.builder().keyType(KeyType.HASH)
+                                                                        .attributeName(HASH_KEY_NAME)
+                                                                        .build())
+                                             .attributeDefinitions(AttributeDefinition.builder()
+                                                                                      .attributeType(ScalarAttributeType.N)
+                                                                                      .attributeName(HASH_KEY_NAME)
+                                                                                      .build())
+                                             .provisionedThroughput(ProvisionedThroughput.builder()
+                                                                                         .readCapacityUnits(5L)
+                                                                                         .writeCapacityUnits(5L)
+                                                                                         .build())
+                                             .build());
+
+        TableUtils.waitUntilActive(dynamo, TABLE_NAME);
+
+        putTestData();
+    }
+
+    @AfterClass
+    public static void cleanUpFixture() {
+        dynamo.deleteTable(DeleteTableRequest.builder().tableName(TABLE_NAME).build());
+    }
+
+    @Test
+    public void test_MultipleIteration_On_Responses_Iterable() {
+        ScanRequest request = ScanRequest.builder().tableName(TABLE_NAME).limit(2).build();
+        ScanPaginator scanResponses = dynamo.scanIterable(request);
```

Comment:
Seeing it definitely don't know if I like "scanIterable"

Reply:
I don't think we will go with "Iterable". We would want to have consistent naming across sync and Async clients. Based on #185, Iterable doesn't make sense for Async operations. So we need change them for sure.